crates/sui-types/src/collection_types.rs

```diff
@@ -26,70 +25,30 @@ pub struct VecSet<T> {
 }
 
 /// Rust version of the Move sui::table::Table type.
-#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+#[derive(Debug, Default, Serialize, Deserialize, Clone, Eq, PartialEq)]
 pub struct TableVec {
     pub contents: Table,
 }
 
-impl Default for TableVec {
-    fn default() -> Self {
-        TableVec {
-            contents: Table {
-                id: ObjectID::from(SuiAddress::ZERO),
-                size: 0,
-            },
-        }
-    }
-}
-
 /// Rust version of the Move sui::table::Table type.
-#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+#[derive(Debug, Default, Serialize, Deserialize, Clone, Eq, PartialEq)]
 pub struct Table {
-    pub id: ObjectID,
+    pub id: ID,
```

Comment:
I'm not so sure about this change.  Ideally all the `id` fields should have type `UID` -- `ObjectID` is okay as well, but not quite right from the perspective of documenting the Move interface for these types.  What's the motivation for this change?  It seems strange to say you can have a `Default` implementation of this type, because there is no such thing as a "default" identity (UID or ObjectID) -- what does it mean to take a default `Table` and read children from it...?